libhb/hwaccel.c

```diff
@@ -136,23 +250,48 @@ int hb_hwaccel_hwframes_ctx_init(AVCodecContext *ctx, hb_job_t *job)
 {
     if (!ctx->hw_device_ctx)
     {
-        hb_error("hwaccel: failed to initialize hw frames context");
+        hb_error("hwaccel: failed to initialize hw frames context - no hw_device_ctx");
         return 1;
     }
 
     ctx->get_format = hw_hwaccel_get_hw_format;
     ctx->pix_fmt = job->hw_pix_fmt;
+#if HB_PROJECT_FEATURE_QSV
+    if (hb_hwaccel_is_full_hardware_pipeline_enabled(job) &&
+            hb_qsv_decode_is_enabled(job))
+    {
+        ctx->extra_hw_frames = 32;
+        ctx->sw_pix_fmt = job->input_pix_fmt;
+    }
+#endif
+
     ctx->hw_frames_ctx = av_hwframe_ctx_alloc(ctx->hw_device_ctx);
 
     AVHWFramesContext *frames_ctx = (AVHWFramesContext *)ctx->hw_frames_ctx->data;
     frames_ctx->format = job->hw_pix_fmt;
-    frames_ctx->sw_format = job->output_pix_fmt;
```

Comment:
Was this being used by the others hw decoders?

Reply:
reverted back for safety